### PR TITLE
Create cut delta

### PR DESCRIPTION
**Pull request by nitzanashi**
## [5.0.0] - Unreleased

- Added full history between RC and Release
- Removed coupling to Changelog
- Removed next release update

## Examples
- https://github.com/zattoo/web-shadow/pull/468
- https://github.com/zattoo/web-shadow/pull/460
- https://github.com/zattoo/web-shadow/pull/466

## Try it
https://github.com/zattoo/web-shadow/actions/workflows/cut-delta.yml

## Ticket
- https://zattoo2.atlassian.net/browse/WEBCORE-111

### CODE REVIEW

**Comment by gotbahn** [comment]
Oh my, oh my…

File: src/git.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ */
+let initialized = false;
+
+/**
+ *
+ * @param {string} command
+ * @returns {string}
+ */
+const execSyncToString = (command) =>{
+    console.log(`>> ${command}`);
```
Comment (gotbahn) [inline src/git.js:14]: do we need log?

File: src/git.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    return execSync(command).toString().replace(/(\r\n|\n|\r| )/gm, '');
+};
+
+export const setUser = () => {
+    execSync('git config user.name "GitHub Actions Bot"');
+    execSync('git config user.email "<>"');
```
Comment (gotbahn) [inline src/git.js:21]: you can set it to webdev@zattoo.com

File: src/git.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    const previousReleaseBranch = execSyncToString(`git branch -r --list '**/release/${project}/**' | tail -1`);
+
+    // return initial commit on main branch as fallback
+    if (!previousReleaseBranch) {
+        return execSyncToString(`git rev-list --max-parents=0 origin/${defaultBranch} -1`);
+    }
+
+    const baseCommit = execSyncToString(`git merge-base origin/${defaultBranch} ${previousReleaseBranch}`);
+
+    console.log(`base commit for the upcoming release is: ${baseCommit}`);
```
Comment (gotbahn) [inline src/git.js:46]: do we need this log?

**Review by gotbahn — COMMENTED** [review]
Wait… why we remove next version raise?

File: src/git.js (hunk 2 above)
Comment (nitzanashi) [reply src/git.js:21]: what for?

File: src/git.js (hunk 3 above)
Comment (nitzanashi) [reply src/git.js:46]: > Wait… why we remove next version raise?

We are gonna be version-less, no version on main branch, freedom

**Comment by nitzanashi** [comment]
> Wait… why we remove next version raise?

We are gonna be version-less, no version on main branch, freedom

File: src/git.js (hunk 2 above)
Comment (gotbahn) [reply src/git.js:21]: It can associate git activity with bot account, not sure if needed, tho

**Comment by gotbahn** [comment]
Let’s have a small ketchup on this after lunch?

File: src/index.js [hunk 4]
```diff
@@ -287,7 +134,7 @@ const createReleaseCandidatePullRequest = async ({
         owner,
         repo,
         title: `Release ${releaseVersion}-${project}`,
-        body: `## Changelog\n\n${versionBody}\n\n`,
+        body: `## TBD`,
```
Comment (gotbahn) [inline src/index.js:137]: need to be removed

File: src/git.js (hunk 2 above)
Comment (gotbahn) [reply src/git.js:21]: probably we don't need setup any user

File: src/git.js (hunk 3 above)
Comment (gotbahn) [reply src/git.js:46]:
```suggestion
    console.log(`Base commit for the upcoming release is: ${baseCommit}`);
```

File: src/index.js [hunk 5]
```diff
@@ -226,7 +86,7 @@ const createReleaseCandidatePullRequest = async ({
             owner,
             repo,
             branch: releaseBranch,
-            sha: baseSha,
+            sha: gitService.getBaseCommit(project, defaultBranch),
```
Comment (mxkyb) [inline src/index.js:89]: can we make a more meaningful difference in the naming between `getBaseCommit` and `baseSha`? Maybe:
`getBaseCommit` --> `getLastCutBaseCommit` 
`baseSha` --> `currentHeadSha`

File: src/git.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+ * @param {string} command
+ * @returns {string}
+ */
+const execSyncToString = (command) =>{
+    console.log(`>> ${command}`);
+
+    return execSync(command).toString().replace(/(\r\n|\n|\r| )/gm, '');
+};
+
+export const setUser = () => {
```
Comment (mxkyb) [inline src/git.js:19]:
```suggestion
export const initGit = () => {
```

File: src/index.js (hunk 5 above)
Comment (nitzanashi) [reply src/index.js:89]: @mxkyb `getBaseCommit` is not gonna give the last `cut`, but rather the last commit on the latest release, so the last point on the main branch was released to our users.

`baseSha` was already there I don't wanna change occurrences of that in this PR

File: src/git.js (hunk 2 above)
Comment (nitzanashi) [reply src/git.js]: Removed

**Review by gotbahn — APPROVED** [review]

**Pull request merged**